app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -3524,10 +3583,44 @@ public String createSnapshot(String location) throws RegistryStorageException {
         return null;
     }
 
+    @Override
+    public String createEvent(OutboxEvent event) {
+        if (supportsDatabaseEvents()) {
+            // Create outbox event
+            handles.withHandle(handle -> handle.createUpdate(sqlStatements.createOutboxEvent())
+                    .bind(0, event.getId()).bind(1, eventsTopic).bind(2, event.getAggregateId())
+                    .bind(3, event.getType()).bind(4, event.getPayload().toString()).execute());
+
+            deleteEvent(event.getId());
+
+        }
+        return event.getId();
+    }
+
+    @Override
+    public boolean supportsDatabaseEvents() {
+        return isPostgresql() || isMssql();
+    }
+
+    private void deleteEvent(String eventId) {
```

Comment:
I tried to have a pattern in the SQL storage where only public methods would call `handles.withHandle()` - the resulting `handle` instance would get passed as an argument to private methods needing it.  Could you change this to follow that pattern?